### ISSUE TITLE
Not able to create pods anymore because of - "can not retrieve list of objects using index : Index with name namespace does not exists" error k8s API server logs.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#7626]
Rancher Versions: v1.4.0-rc5

K8s environment with 3 AWS hosts with  RancherOS v0.7.1 (4.4.24) wit cpu - 4x2.4 GHz and memory - 15.7 GiB
We are  able create service/rc/pods and automation tests reports success.
After sometime into the run , we see K8s environment becomes unusable. Any K8s command relating to creating pods/namespace starts failing.

At this point , we are not able to launch "Dashboard UI" as well
Following error seen in k8s API server logs

```
1/31/2017 9:39:16 AMW0131 17:39:16.602206       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
```

Also etcd containers show this error:
```
1/31/2017 9:18:34 AM2017-01-31 17:18:34.346251 W | etcdserver: failed to send out heartbeat on time (deadline exceeded for 214.880079ms)
1/31/2017 9:18:34 AM2017-01-31 17:18:34.346255 W | etcdserver: server is likely overloaded
```

Restarting K8s API server , will get us out of this issue and K8s environment becomes useable.

**Comment by sangeethah** [comment]
@galal-hussein: This is the same issue you have seen in your setup that was also seen when testing with 1.3.3.

**Comment by LLParse** [comment]
@sangeethah Etcd is being starved of resources it needs to function, consequently it stops accepting write requests. There are 2 solutions:
1. (exists today) Run etcd on dedicated hosts via labels `etcd=true` and `nopods=true`
2. (new) Update etcd template to make heartbeat period configurable. We still may occasionally experience these errors, but it would be fixable by upgrading the service and increasaing the heartbeat period.

A final option that we've decided not to pursue on countless occasions is dedicating CPU/mem resources to etcd, so I did not include it in the list.

**Comment by sangeethah** [comment]
Tested with build from master using docker version 1.13.1 on set up that has 3 hosts that have 4 GB memory.

<img width="896" alt="screen shot 2017-02-10 at 9 49 51 am" src="https://cloud.githubusercontent.com/assets/4266958/22837675/6789528c-ef76-11e6-87d1-52abb5ea74a0.png">


When automation was run , few cases succeed after which all cases failed. Lot of these errors are seen in logs:

I also see this 1 line in particular . Not sure what this means.
```
2/10/2017 8:42:58 AMI0210 16:42:58.490617       1 etcd_helper.go:191] deletion of /registry/controllers/rollingupdates-namespace/testru failed because of a conflict, going to retry
```

After the automation run , I am able to use the set up now (create namespace and pods) .But continue to see these errors in the logs:

```
2/10/2017 8:38:52 AM[858.336908ms] [29.431µs] Node list decoded
2/10/2017 8:38:52 AM[858.476772ms] [139.864µs] END
2/10/2017 8:38:52 AMI0210 16:38:52.812140       1 trace.go:61] Trace "List /apis/batch/v2alpha1/cronjobs" (started 2017-02-10 16:38:51.952708229 +0000 UTC):
2/10/2017 8:38:52 AM[16.059µs] [16.059µs] About to List from storage
2/10/2017 8:38:52 AM[858.911594ms] [858.895535ms] Listing from storage done
2/10/2017 8:38:52 AM[859.037019ms] [125.425µs] Self-linking done
2/10/2017 8:38:52 AM[859.25678ms] [219.761µs] Writing http response done (0 items)
2/10/2017 8:38:52 AM[859.25939ms] [2.61µs] END
2/10/2017 8:39:39 AMW0210 16:39:39.349164       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:39:39 AMW0210 16:39:39.352732       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:39:43 AMW0210 16:39:43.951313       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:40:26 AMW0210 16:40:26.001205       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:40:26 AMW0210 16:40:26.003758       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:40:29 AMI0210 16:40:29.769045       1 trace.go:61] Trace "etcdHelper::Create *api.Event" (started 2017-02-10 16:40:29.300189387 +0000 UTC):
2/10/2017 8:40:29 AM[74.833µs] [74.833µs] Object encoded
2/10/2017 8:40:29 AM[77.002µs] [2.169µs] Version checked
2/10/2017 8:40:29 AM[468.642944ms] [468.565942ms] Object created
2/10/2017 8:40:29 AM[468.784588ms] [141.644µs] END
2/10/2017 8:41:26 AMW0210 16:41:26.763939       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:41:53 AMW0210 16:41:53.357283       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:41:54 AMW0210 16:41:54.115975       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:42:58 AMI0210 16:42:58.490617       1 etcd_helper.go:191] deletion of /registry/controllers/rollingupdates-namespace/testru failed because of a conflict, going to retry
2/10/2017 8:43:56 AMW0210 16:43:56.296072       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.312107       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.316382       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.381356       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.385321       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.384537       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.396131       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.396760       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.397683       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.410188       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.425068       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.427660       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.483111       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.487492       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.488960       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.806916       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.808512       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.813200       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.986707       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.988385       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:56 AMW0210 16:43:56.994964       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:57 AMW0210 16:43:57.320145       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:57 AMW0210 16:43:57.322703       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:57 AMW0210 16:43:57.323843       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:57 AMW0210 16:43:57.977429       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:57 AMW0210 16:43:57.978112       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:57 AMW0210 16:43:57.981066       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:59 AMW0210 16:43:59.276759       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:59 AMW0210 16:43:59.280306       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:43:59 AMW0210 16:43:59.281321       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:01 AMW0210 16:44:01.852910       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:01 AMW0210 16:44:01.853268       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:01 AMW0210 16:44:01.854272       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:06 AMW0210 16:44:06.984019       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:06 AMW0210 16:44:06.984037       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:06 AMW0210 16:44:06.985006       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:17 AMW0210 16:44:17.238178       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:17 AMW0210 16:44:17.241704       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:17 AMW0210 16:44:17.243745       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:37 AMW0210 16:44:37.728770       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:37 AMW0210 16:44:37.731403       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:44:37 AMW0210 16:44:37.732528       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:45:18 AMW0210 16:45:18.707405       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:45:18 AMW0210 16:45:18.712209       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:45:18 AMW0210 16:45:18.713506       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:46:40 AMW0210 16:46:40.644451       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:46:40 AMW0210 16:46:40.648363       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:46:40 AMW0210 16:46:40.648408       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:49:24 AMW0210 16:49:24.511755       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:49:24 AMW0210 16:49:24.512868       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:49:24 AMW0210 16:49:24.516272       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:53:40 AMW0210 16:53:40.805057       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:53:40 AMW0210 16:53:40.807447       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:53:40 AMW0210 16:53:40.808720       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:54:52 AMW0210 16:54:52.213275       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:54:52 AMW0210 16:54:52.217872       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 8:54:52 AMW0210 16:54:52.221436       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:03:40 AMW0210 17:03:40.809624       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:03:40 AMW0210 17:03:40.814908       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:03:40 AMW0210 17:03:40.818071       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:11:32 AMW0210 17:11:32.232373       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:11:32 AMW0210 17:11:32.235088       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:11:32 AMW0210 17:11:32.238700       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:13:40 AMW0210 17:13:40.808639       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:13:40 AMW0210 17:13:40.810713       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:13:40 AMW0210 17:13:40.811948       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:23:40 AMW0210 17:23:40.815727       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:23:40 AMW0210 17:23:40.815636       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:23:40 AMW0210 17:23:40.822000       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:28:12 AMW0210 17:28:12.259441       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:28:12 AMW0210 17:28:12.260988       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:28:12 AMW0210 17:28:12.263691       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:33:40 AMW0210 17:33:40.811189       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:33:40 AMW0210 17:33:40.811530       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:33:40 AMW0210 17:33:40.812031       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:43:40 AMW0210 17:43:40.813720       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:43:40 AMW0210 17:43:40.815892       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:43:40 AMW0210 17:43:40.819854       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:44:52 AMW0210 17:44:52.287761       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:44:52 AMW0210 17:44:52.291430       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:44:52 AMW0210 17:44:52.292803       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:47:36 AMW0210 17:47:36.883935       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
2/10/2017 9:47:36 AMW0210 17:47:36.884087       1 listers.go:69] can not retrieve list of objects using index : Index with name namespace does not exist
```

**Comment by LLParse** [comment]
https://github.com/rancher/rancher-catalog/pull/452

**Comment by LLParse** [comment]
Available in rancher-catalog `v1.5.0-dev` branch.

**Comment by galal-hussein** [comment]
Rancher version 1.5.0-alpha1

steps to validate:
- Install Rancher server
- Deploy kubernetes with a little value of heartbeat 5 ms or 10 ms
- Make sure that etcd is throwing heartbeat error.
After running automation, i can see a lot of failures.

- Upgrade kubernetes with a big heartbeat value.

I can no longer see heartbeat errors.
